core/templates/dev/head/services/UserServiceSpecs.js

```diff
@@ -29,6 +28,20 @@ describe('User Service', function() {
 
   it('should return image data', function() {
     var requestUrl = '/preferenceshandler/profile_picture';
+    // creating a test user for checking profile picture of user.
+    var sampleUserInfoBackendObject = {
+      is_moderator: false,
+      is_admin: false,
+      is_super_admin: false,
+      is_topic_manager: false,
+      can_create_collections: true,
+      preferred_site_language_code: null,
+      username: 'tester',
+      user_is_logged_in: true
+    };
+    $httpBackend.when('GET', '/userinfohandler').respond(200,
```

Comment:
@vojtechjelinek , I am using `$httpBackend.when` here. Here, I want to make a dummy user and login to oppia so that other test cases work perfectly but when I use `$httpBackend.expect` it gives me error ( `No pending request to flush` ). Is there any workaround?

Reply:
On what line it gives you that error, maybe copy the whole error here?

Reply:
```
Chrome 73.0.3683 (Linux 0.0.0) User Service should return image data FAILED
	Error: No pending request to flush !
	    at Function.$httpBackend.flush (third_party/static/angularjs-1.5.8/angular-mocks.js:1799:34)
	    at Object.<anonymous> (core/templates/dev/head/services/UserServiceSpecs.js:9:2193)
```

Here is a screenshot


![Error-oppia](https://user-images.githubusercontent.com/12140464/56227354-42b8e680-6043-11e9-8d22-7b017f1112e0.png)

Reply:
As I think of it you don't need to use `.expect()` for the `/userinfohandler` using when is okay, but if you use `.expect()` you shouldn't need to add any new `.flush()` the flush on line 52 should be enough.

Reply:
Okay!. This PR is ready for review. PTAL. Thanks!